### PR TITLE
feat: situations and notices in departure details

### PR DESCRIPTION
**Pull request by mortennordseth**
This PR features situations and notices on the departure detail page. 

At the time of writing this, one can look up bus 78 from Ringvål and from Lergårdsbakken to test situations. 

Closes https://github.com/AtB-AS/kundevendt/issues/9162

<details>
<summary>Screenshots</summary>

image

image

image

</details>

### CODE REVIEW

**Comment by mortennordseth** [comment]
We need to find a good solution for handling accessibility for texts, but I'm not sure if that should be done in https://github.com/AtB-AS/kundevendt/issues/15431 and prioritize it.

**Comment by adriansberg** [comment]
> accessibility for texts

What does this include?

File: src/modules/situations/situation-message-box.tsx [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+import { Situation } from './types';
+import { useTranslation, getTextForLanguage } from '@atb/translations';
+import { useTheme } from '@atb/modules/theme';
+import style from './situation.module.css';
+import { Button } from '@atb/components/button';
+import { useRef } from 'react';
+import dictionary from '@atb/translations/dictionary';
+import { Situation as SituationTexts } from '@atb/translations/modules';
+import { formatToLongDateTime } from '@atb/utils/date';
+export type Props = {
```
Comment (adriansberg) [inline src/modules/situations/situation-message-box.tsx:20]:
```suggestion

export type Props = {
```

File: src/modules/situations/situation.module.css [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+.messageBox__button {
+  padding: var(--spacings-small);
+}
+.icon {
+  margin-top: var(--spacings-small);
+}
+
+.dialog {
+  padding: var(--spacings-large);
+  margin: auto auto;
```
Comment (adriansberg) [inline src/modules/situations/situation.module.css:30]:
```suggestion
  margin: auto;
```

File: src/modules/situations/situation.module.css [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  border: none;
+  border-radius: var(--border-radius-circle);
+  background-color: var(--static-background-background_1-background);
+}
+
+.dialog::backdrop {
```
Comment (adriansberg) [inline src/modules/situations/situation.module.css:36]: TIL. Did not know that this was a pseudo selector.

**Comment by mortennordseth** [comment]
> What does this include?

We've been chatting about how to make the screenreaders handle headers and similar information, `Bus 22 from Strinda vgs`, for example. This also applies to the situation/notice message box information

File: src/page-modules/departures/details/index.tsx [hunk 4]
```diff
@@ -58,11 +71,44 @@ export function DeparturesDetails({
...
+) => {
+  const focusedEstimatedCall =
+    serviceJourney.estimatedCalls?.find(
+      ({ quay }) => quay?.id && quay.id === fromQuayId,
+    ) || serviceJourney.estimatedCalls?.[0];
+
+  return filterNotices([
+    ...serviceJourney.notices,
+    ...serviceJourney.line.notices,
+    ...(focusedEstimatedCall?.notices || []),
```
Comment (adriansberg) [inline src/page-modules/departures/details/index.tsx:112]:
```suggestion
    ...(focusedEstimatedCall?.notices ?? []),
```

File: src/page-modules/departures/details/index.tsx [hunk 5]
```diff
@@ -58,11 +71,44 @@ export function DeparturesDetails({
...
 }
+
+export const getNoticesForServiceJourney = (
+  serviceJourney: ServiceJourneyData,
+  fromQuayId?: string,
+) => {
+  const focusedEstimatedCall =
+    serviceJourney.estimatedCalls?.find(
+      ({ quay }) => quay?.id && quay.id === fromQuayId,
+    ) || serviceJourney.estimatedCalls?.[0];
```
Comment (adriansberg) [inline src/page-modules/departures/details/index.tsx:107]:
```suggestion
    ) ?? serviceJourney.estimatedCalls?.[0];
```

**Review by adriansberg — APPROVED** [review]
Great work!

File: src/modules/situations/situation.module.css (hunk 3 above)
Comment (mortennordseth) [reply src/modules/situations/situation.module.css:36]: Jacob 🤝 🎉

**Pull request merged**